media-gfx/blender/blender-4.0.2.ebuild

```diff
@@ -0,0 +1,483 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..11} )
+
+inherit check-reqs cmake cuda flag-o-matic pax-utils python-single-r1 toolchain-funcs xdg-utils
+
+DESCRIPTION="3D Creation/Animation/Publishing System"
+HOMEPAGE="https://www.blender.org"
+
+if [[ ${PV} = *9999* ]] ; then
+	# Subversion is needed for downloading unit test files
+	inherit git-r3 subversion
+	EGIT_REPO_URI="https://projects.blender.org/blender/blender.git"
+	ADDONS_EGIT_REPO_URI="https://projects.blender.org/blender/blender-addons.git"
+else
+	SRC_URI="https://download.blender.org/source/${P}.tar.xz"
+	# Update these between major releases.
+	TEST_TARBALL_VERSION="$(ver_cut 1-2).0"
+	# SRC_URI+=" test? ( https://dev.gentoo.org/~sam/distfiles/${CATEGORY}/${PN}/${PN}-${TEST_TARBALL_VERSION}-tests.tar.xz )"
+	KEYWORDS="~amd64 ~arm ~arm64"
+fi
+
+SLOT="${PV%.*}"
+LICENSE="|| ( GPL-3 BL )"
+IUSE="+bullet +fluid +openexr +tbb
+	alembic collada +color-management cuda +cycles cycles-bin-kernels
+	debug doc +embree +ffmpeg +fftw +gmp jack jemalloc jpeg2k
+	man +nanovdb ndof nls openal +oidn +openmp +openpgl +opensubdiv
+	+openvdb optix osl +pdf +potrace +pugixml pulseaudio sdl
+	+sndfile test +tiff valgrind wayland +webp X"
+RESTRICT="test"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}
+	alembic? ( openexr )
+	cuda? ( cycles )
+	cycles? ( openexr tiff )
```

Comment:
Could you set 
```
	cycles? ( openexr tiff tbb )
	openvdb? ( tbb openexr )
	osl? ( cycles pugixml openexr )
```
Otherwise combination USE="-tbb cycles" silently compiles without cycles. Ref: https://github.com/blender/blender/blob/main/CMakeLists.txt#L1135-L1151

Reply:
Hmm, perhaps we should also turn on `WITH_STRICT_BUILD_OPTIONS` to prevent things like this silently happening in the future as well?

Reply:
Oh, great option, please enable it. If this option didn't exist, I'd suggest even applying a patch like it was in caffe2: https://github.com/gentoo/gentoo/blob/56aefda1a0bee3494c1d4fc844d53074fc404500/sci-libs/caffe2/files/caffe2-2.1.2-rocm-gentoo.patch#L4-L10